app/controller.go

```diff
@@ -89,6 +98,53 @@ func (fn HTTPHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+const headerContentType = "Content-Type"
+
+type HostForwarder struct {
+	AddressResolver HostAddressResolver
+	Client          *http.Client
+}
+
+func (f *HostForwarder) Handler() HTTPHandler {
+	return func(w http.ResponseWriter, r *http.Request) error {
+		vars := mux.Vars(r)
+		zone := vars["zone"]
+		if zone == "" {
+			return NewBadRequestError("Invalid url, missing zone value", nil)
+		}
+		host := vars["host"]
+		if host == "" {
+			return NewBadRequestError("Invalid url, missing host value", nil)
+		}
+		split := strings.SplitN(r.URL.String(), "hosts/"+host+"/", 2)
+		if len(split) != 2 {
+			return NewBadRequestError("Invalid url, missing host resource", nil)
```

Comment:
Is this possible? The matcher specified in the route should guarantee that the URL contains a non empty prefix before the separator string and a non empty suffix after it. Which means this should be an assertion turning into a server error (missconfigured server routes) when found instead of a bad request from the user.

Reply:
good catch, totally agree. done!